src/DynamoCore/Extensions/ExtensionManager.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Dynamo.Extensions
+{
+    public class ExtensionManager: IExtensionManager
+    {
+        List<IExtension> extensions = new List<IExtension>();
+        IExtensionLoader extensionLoader = new ExtensionLoader();
+
+        public void Add(IExtension extension)
+        {
+            if (extensions.Find(ext => ext.UniqueId == extension.UniqueId) == null)
+            {
+                extensions.Add(extension);
+
+                if (ExtensionAdded != null)
+                {
+                    ExtensionAdded(extension);
+                }
+            }
```

Comment:
If the `IExtension` is a duplicate, let's send an appropriate message to the log.